### PR TITLE
Error logging improvement

### PR DESCRIPTION
**Pull request by sareh**
No ticket

**Overall change:** 
Updates data fetch error logging to have JSON format of this type:

```
2020-03-07 11:24:36 error [fetchPageData/index.js] {
  "event": "data_response_404",
  "status": 404,
  "message": "Data not found when requesting http://localhost:7080/persian/articles/c1vlle3q337o.json",
  "url": "http://localhost:7080/persian/articles/c1vlle3q337o.json"
}
```
event `data_response_404` for when the data requested has a response status of 404.
event `data_fetch_error` for when the data fetch errors.

Following the structure added for other error logs carried out in this PR: https://github.com/bbc/simorgh/pull/5784/files for errors on the server render. 

Updated other errors so they are printed in prettified way for easy scanning in sumologic logs (currently setup was just a single line with all of this concatenated)

**Before**
```
2020-03-07 11:40:14 info [server/index.jsx] {"event":"ssr_request_received","url":"/persian","urlPath":"/persian","headers":{"host":"localhost:7080","user-agent":"...","accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8","accept-language":"en-GB,en;q=0.5","accept-encoding":"gzip, deflate","dnt":"1","connection":"keep-alive","upgrade-insecure-requests":"1"}}

```
**After**
```
2020-03-07 11:38:36 info [server/index.jsx] {
  "event": "ssr_request_received",
  "url": "/persian",
  "urlPath": "/persian",
  "headers": {
    "host": "localhost:7080",
    "user-agent": ".....",
    "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
    "accept-language": "en-GB,en;q=0.5",
    "accept-encoding": "gzip, deflate",
    "dnt": "1",
    "connection": "keep-alive",
    "upgrade-insecure-requests": "1"
  }
}
```

**Code changes:**
- event `data_response_404` for when the data requested has a response status of 404.
- event `data_fetch_error` for when the data fetch errors.
- updated other errors so they are printed in prettified way for easy scanning in sumologic logs.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,7 +18,16 @@ let currentApp = app;
 
 server.listen(port, error => {
   if (error) {
-    logger.error(error);
+    logger.error(
+      JSON.stringify(
+        {
+          event: 'server_listen_error',
+          message: error.toString(),
```
Comment (sareh) [inline src/index.js]:
```suggestion
          message: error,
```

File: src/server/index.jsx [hunk 2]
```diff
@@ -114,7 +114,16 @@ if (process.env.SIMORGH_APP_ENV !== 'local') {
 const sendDataFile = (res, dataFilePath, next) => {
   res.sendFile(dataFilePath, {}, sendErr => {
     if (sendErr) {
-      logger.error(sendErr);
+      logger.error(
+        JSON.stringify(
+          {
+            event: 'local_sendfile_error',
+            message: sendErr.toString(),
```
Comment (sareh) [inline src/server/index.jsx]:
```suggestion
            message: sendErr,
```

File: src/server/index.jsx [hunk 3]
```diff
@@ -219,7 +228,16 @@ server
     const swPath = `${__dirname}/public/sw.js`;
     res.sendFile(swPath, {}, error => {
       if (error) {
-        logger.error(error);
+        logger.error(
+          JSON.stringify(
+            {
+              event: 'server_sendfile_error_sw',
+              message: error.toString(),
```
Comment (sareh) [inline src/server/index.jsx]:
```suggestion
              message: error,
```

**Review by chris-hinds — APPROVED** [review]
Thanks for this 👍

**Comment by chris-hinds** [comment]
The failing tests are due to the logging output being changed. 

Maybe its not worth us ensuring a specific error message is logged, instead just checking the logger was called?

**Review by tochwill — APPROVED** [review]

**Pull request merged**